src/com/twitter/intellij/pants/execution/PantsMakeBeforeRun.java

```diff
@@ -111,6 +112,14 @@ else if (runConfiguration instanceof CommonProgramRunConfigurationParameters) {
       if (buildRoot.isPresent()) {
         ((CommonProgramRunConfigurationParameters) runConfiguration).setWorkingDirectory(buildRoot.get().getPath());
       }
+
+      if(runConfiguration instanceof JUnitConfiguration) {
+        Optional<String[]> os = PantsOptions.getPantsOptions(project)
+          .flatMap(options -> options.getList(PantsConstants.PANTS_OPTION_TEST_JUNIT_OPTIONS));
+        String optionsString = String.join(" ", os.orElse(new String[]{}));
+        ((JUnitConfiguration)runConfiguration).setVMParameters(optionsString);
+      }
+
```

Comment:
This logic is needed for `if (PantsUtil.isScalaRelatedTestRunConfiguration(runConfiguration))` as well. So possibly we can extract this to a helper function.

Reply:
Done